File: common_ui/src/main/java/com/xiaoyv/ui/dialog/OptionsDialogItemBinder.java
package com.xiaoyv.ui.dialog;

import android.graphics.Typeface;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.ColorInt;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.blankj.utilcode.util.ColorUtils;
import com.blankj.utilcode.util.LogUtils;
import com.drakeet.multitype.ItemViewBinder;
import com.xiaoyv.ui.R;
import com.xiaoyv.ui.databinding.UiDialogOptionsItemBinding;

import org.jetbrains.annotations.NotNull;

/**
 * OptionsDialog
 *
 * @author why
 * @since 2020/12/01
 **/
public class OptionsDialogItemBinder extends ItemViewBinder<String, OptionsDialogItemBinder.ViewHolder> {
    private OnItemChildClickListener clickListener;
    @ColorInt
    private int textColor;
    @ColorInt
    private int lastTextColor;
    private int textSize;
    private Typeface textStyle;

    public OptionsDialogItemBinder() {
        textSize = 16;
        textStyle = Typeface.DEFAULT;
        textColor = ColorUtils.getColor(R.color.ui_text_c1);
        lastTextColor = ColorUtils.getColor(R.color.ui_text_c1);
    }

    public void setOnItemChildClickListener(OnItemChildClickListener clickListener) {
        this.clickListener = clickListener;
    }

    public void setTextColor(int textColor) {
        this.textColor = textColor;
    }

    public void setLastTextColor(int lastTextColor) {
        this.lastTextColor = lastTextColor;
    }

    public void setTextSize(int textSize) {
        this.textSize = textSize;
    }

    public void setTextStyle(Typeface textStyle) {
        this.textStyle = textStyle;
    }


    @NotNull
    @Override
    public ViewHolder onCreateViewHolder(@NotNull LayoutInflater layoutInflater, @NotNull ViewGroup viewGroup) {
        UiDialogOptionsItemBinding itemBinding = UiDialogOptionsItemBinding.inflate(layoutInflater, viewGroup, false);
        return new ViewHolder(itemBinding.getRoot(), itemBinding);
    }

    @Override
    public void onBindViewHolder(@NotNull ViewHolder viewHolder, String option) {
        viewHolder.itemBinding.tvOptions.setText(option);
        viewHolder.itemBinding.tvOptions.setTypeface(textStyle);
        viewHolder.itemBinding.tvOptions.setTextSize(textSize);
        viewHolder.itemBinding.tvOptions.setOnClickListener(v -> {
            if (this.clickListener != null) {
                this.clickListener.onItemChildClick(getPosition(viewHolder));
            }
        });

        if (getPosition(viewHolder) == getAdapterItems().size() - 1) {
            viewHolder.itemBinding.uiView.setVisibility(View.GONE);
            viewHolder.itemBinding.tvOptions.setTextColor(lastTextColor);
        } else {
            viewHolder.itemBinding.uiView.setVisibility(View.VISIBLE);
            viewHolder.itemBinding.tvOptions.setTextColor(textColor);
        }
    }

    /**
     * ViewHolder
     */
    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final UiDialogOptionsItemBinding itemBinding;

        public ViewHolder(@NonNull View itemView, UiDialogOptionsItemBinding itemBinding) {
            super(itemView);
            this.itemBinding = itemBinding;
        }
    }

    public interface OnItemChildClickListener {
        /**
         * 选项点击事件
         *
         * @param position 事件
         */
        void onItemChildClick(int position);
    }
}

File: common_ui/src/main/java/com/xiaoyv/ui/dialog/OptionsDialogItemBinder.kt
package com.xiaoyv.ui.dialog

import android.graphics.Typeface
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.ColorInt
import com.blankj.utilcode.util.ColorUtils
import com.xiaoyv.ui.R
import com.xiaoyv.ui.base.BaseItemBindingBinder
import com.xiaoyv.ui.databinding.UiDialogOptionsItemBinding

/**
 * OptionsDialog
 *
 * @author why
 * @since 2020/12/01
 */
class OptionsDialogItemBinder : BaseItemBindingBinder<String, UiDialogOptionsItemBinding>() {
    var clickListener: OnItemChildClickListener? = null

    @ColorInt
    var textColor: Int = ColorUtils.getColor(R.color.ui_text_c1)

    @ColorInt
    var lastTextColor: Int = ColorUtils.getColor(R.color.ui_text_c1)

    var textSize = 14f

    var textStyle: Typeface = Typeface.DEFAULT

    override fun onCreateViewBinding(
        layoutInflater: LayoutInflater,
        parent: ViewGroup,
        viewType: Int
    ): UiDialogOptionsItemBinding {
        return UiDialogOptionsItemBinding.inflate(layoutInflater, parent, false)
    }

    override fun convert(
        holder: BinderVBHolder<UiDialogOptionsItemBinding>,
        binding: UiDialogOptionsItemBinding,
        data: String
    ) {
        binding.tvOptions.text = data
        binding.tvOptions.typeface = textStyle
        binding.tvOptions.textSize = textSize
        binding.tvOptions.setOnClickListener {
            clickListener?.onItemChildClick(holder.bindingAdapterPosition)
        }
        if (holder.bindingAdapterPosition == adapter.data.size - 1) {
            binding.uiView.visibility = View.GONE
            binding.tvOptions.setTextColor(lastTextColor)
        } else {
            binding.uiView.visibility = View.VISIBLE
            binding.tvOptions.setTextColor(textColor)
        }
    }

    interface OnItemChildClickListener {
        /**
         * 选项点击事件
         *
         * @param position 事件
         */
        fun onItemChildClick(position: Int)
    }
}